### DISCUSSION TITLE
How to enable upstream proxies

### DISCUSSION THREAD

**Discussion by Fireant456** [FlareSolverr/FlareSolverr#1089] [Q&A]
Hello,

I would like to proxy my traffic after going through flaresolverr through a VPN Privoxy Proxy. I see there is a configuration for proxies in the commands but I would like to send all traffic after flaresolverr through the Privoxy proxy. How can I set this up?

**Comment by ilike2burnthing** [comment]
Set up the proxy at a system level.

**Reply by Fireant456** [reply]
I use Flaresolverr in docker, is there a way to do this still?

**Reply by Fireant456** [reply]
Also can you please not mark this as closed for me? I can mark it as answered once my issue is resolved :/

**Reply by ilike2burnthing** [reply]
Either run a proxy/VPN on your host device that Docker is running on, or add a proxy/VPN container and direct the traffic for FlareSolverr and any other relevant containers (e.g. Jackett/Prowlarr) through it.

However, if you're just communicating between two containers on the same device, this is completely unnecessary.

**Reply by Fireant456** [reply]
I have traffic that I want going over my VPN proxy also going through FlareSolverr for the websites that have CloudFlare challenges...

So my flow would be:

Source container making API query -> FlareSolverr via HTTP proxy -> VPN Privoxy HTTP proxy

I am looking for a way to set my FlareSolverr container to use the VPN proxy for its connection, but from what I can tell there is no native way to do this with docker so I would need the container to support it in some fashion. Do you know some way to do this that I am not finding? I would rather not change up my VPN setup and build a VPN container or complicated network configuration for my FlareSolverr container when all my other containers requiring VPN can simply utilize the proxy I already have.

**Reply by ilike2burnthing** [reply]
Either feed the response from FlareSolverr back to the source container, and then send that via the proxied connection, or the options in https://github.com/FlareSolverr/FlareSolverr/discussions/1089#discussioncomment-8551916.
